src/test/java/codeu/controller/LoginServletTest.java

```diff
@@ -71,7 +71,12 @@ public void testDoPost_NewUser() throws IOException, ServletException {
   @Test
   public void testDoPost_ExistingUser() throws IOException, ServletException {
 
-    User user = new User(UUID.randomUUID(), "test username", "test password", Instant.now());
+    User user =
+        new User(
+            UUID.randomUUID(),
+            "test username",
+            "$2a$10$.e.4EEfngEXmxAO085XnYOmDntkqod0C384jOR9oagwxMnPNHaGLa",
+            Instant.now());
 
     Mockito.when(mockRequest.getParameter("username")).thenReturn("test username");
     Mockito.when(mockRequest.getParameter("password")).thenReturn("test password");
```

Comment:
Does this have to change (still uses "test password")?

Reply:
Good question. This one doesn't have to change, because it's the raw unhashed value coming from the user (or in this case, the mock user).

The bcrypt library takes this unhashed password and compares it to the hashed password that we have saved. This happens in `LoginServlet` on this line:

```
if (!BCrypt.checkpw(password, user.getPasswordHash())) {
```

Here, the `password` argument is the unhashed password, which is the `"test password"` on this line represents. BCrypt then does its magic to check for validity against the hashed value we have saved.

The changes I made to the tests are all to places that should be dealing with the hashed password, not the raw unhashed value.

Reply:
Got it. Thanks.